### PR TITLE
Ory component integration test on eval profile

### PR DESCRIPTION
**Pull request by cnvergence**
**Description**

Changes proposed in this pull request:

- New pipeline that will run on changes to Ory component in the reconciler.

**Related issue(s)**

#4656

### CODE REVIEW

File: prow/scripts/provision-vm-and-start-reconciler-component-integration-k3d.sh [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  log::info "Created reconciler-component-integration-test-${RANDOM_ID} in zone ${ZONE}" && break
+  log::error "Could not create machine in zone ${ZONE}"
+done || exit 1
+ENDTIME=$(date +%s)
+echo "VM creation time: $((ENDTIME - STARTTIME)) seconds."
+
+trap cleanup exit INT
+
+log::info "Preparing environment variables for the instance"
+envVars=(
```
Comment (strekm) [inline prow/scripts/provision-vm-and-start-reconciler-component-integration-k3d.sh:111]: this env shouldn't be hardcoded. think about istio tests or other reconciler tests

File: prow/jobs/incubator/reconciler/reconciler.yaml [hunk 2]
```diff
@@ -48,6 +48,53 @@ presubmits: # runs on PRs
...
+        containers:
+          - image: "eu.gcr.io/kyma-project/test-infra/kyma-integration:v20220105-baf0471a"
+            securityContext:
+              privileged: true
+            command:
+              - "/home/prow/go/src/github.com/kyma-project/test-infra/prow/scripts/provision-vm-and-start-reconciler-component-integration-k3d.sh"
+            env:
+              - name: GO11MODULE
+                value: "on"
+              - name: ORY_INTEGRATION
```
Comment (strekm) [inline prow/jobs/incubator/reconciler/reconciler.yaml:85]: consider changing this env to something which would indicate what test is that. something like 
TEST_NAME="ORY_INTEGRATION". this way we don't need to hardcode this env but just pass it to scripts

File: prow/scripts/cluster-integration/reconciler-component-integration.sh [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  - name: "certificates"
+    namespace: "istio-system"
+components:
+  - name: "ory"
+EOF
+}
+
+function deploy_kyma() {
+  local kyma_deploy_cmd
+  kyma_deploy_cmd="kyma deploy -p evaluation --ci"
```
Comment (strekm) [inline prow/scripts/cluster-integration/reconciler-component-integration.sh:72]: kyma cli shouldn't be used in this pipeline. we want to test how reconciler changes influence ory installation

File: prow/scripts/cluster-integration/reconciler-component-integration.sh [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    kyma_deploy_cmd+=" --components-file $PWD/components.yaml"
+  fi
+
+  $kyma_deploy_cmd
+
+  kubectl get pods -A
+}
+
+function run_tests() {
+  echo "Install Go"
```
Comment (strekm) [inline prow/scripts/cluster-integration/reconciler-component-integration.sh:85]: why do we need that here?

File: prow/scripts/cluster-integration/reconciler-component-integration.sh [hunk 5]
```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o pipefail
+
+readonly RECONCILER_DIR="./reconciler"
+readonly GO_VERSION=1.17.5
+export KYMA_SOURCES_DIR="./kyma"
```
Comment (strekm) [inline prow/scripts/cluster-integration/reconciler-component-integration.sh:8]: what version of kyma is installed?

**Review by strekm — CHANGES_REQUESTED** [review]

File: prow/scripts/cluster-integration/reconciler-component-integration.sh (hunk 5 above)
Comment (cnvergence) [reply prow/scripts/cluster-integration/reconciler-component-integration.sh:8]: added KYMA_VERSION

File: prow/scripts/provision-vm-and-start-reconciler-component-integration-k3d.sh (hunk 1 above)
Comment (cnvergence) [reply prow/scripts/provision-vm-and-start-reconciler-component-integration-k3d.sh:111]: we are just passing env from prowjob config to the provisioned VM

File: prow/jobs/incubator/reconciler/reconciler.yaml [hunk 6]
```diff
@@ -48,6 +48,53 @@ presubmits: # runs on PRs
...
+        pipeline.trigger: "pr-submit"
+      labels:
+        prow.k8s.io/pubsub.project: "sap-kyma-prow"
+        prow.k8s.io/pubsub.runID: "pre-main-ory-reconciler-eval-integration"
+        prow.k8s.io/pubsub.topic: "prowjobs"
+        preset-build-pr: "true"
+        preset-gc-project-env: "true"
+        preset-kyma-guard-bot-github-token: "true"
+        preset-sa-vm-kyma-integration: "true"
+      run_if_changed: '^((pkg/reconciler/instances/ory\S+)(\.[^.][^.][^.]+$|\.[^.][^dD]$|\.[^mM][^.]$|\.[^.]$|/[^.]+$))'
```
[the remark this answers is not in the document]
Comment (cnvergence) [reply prow/jobs/incubator/reconciler/reconciler.yaml:63]: hmm, with doing that, wouldn't it be better for us, just to pin to the current reconciler integration tests?
We wouldn't need the 'make test-ory' target then. Our tests would run normally with reconciler integration tests.

**Review by Halamix2 — APPROVED** [review]

**Review by strekm — APPROVED** [review]
/meow

**Pull request merged**